File: insertion-sort/insertion_sort.py
import numpy as np

def insertion_sort(array: np.ndarray) -> np.ndarray:
    for i in range(len(array)):
        value = array[i]
        j = i - 1
        while j >= 0 and array[j] > value:
            array[j + 1] = array[j]
            j -= 1
        array[j + 1] = value
    return array

userinput = input("Enter the array you want to sort with each element seperated by spaces:\n").split()

try:
  print("{", *insertion_sort(np.array([float(x) for x in userinput])), "}")
except:
  print("{", *insertion_sort(np.array([x for x in userinput], dtype=str)), "}")

File: insertion-sort/insertion_sort.cpp
#include <cstdio>

void insertion_sort(int* array, int size) {
    int i, j, value;
    for (i = 0; i < size; i++) {
        value = array[i];
        j = i - 1;
        while (j >= 0 && array[j] > value) {
            array[j + 1] = array[j];
            j--;
        }
        array[j + 1] = value;
    }
}

int main() {
    int i, length;
    std::printf("Length of array to sort: ");
    std::scanf("%d", &length);
    int array1[length];
    std::printf("Enter each element in the array separated by spaces, then click enter:\n");
    
    std::getchar();

    for (i = 0; i < length; i++) {
        std::scanf("%d", &array1[i]);
    }

    std::getchar();

    insertion_sort(array1, sizeof(array1) / sizeof(array1[0]));

    std::printf("{");
    for (int i = 0; i < sizeof(array1) / sizeof(array1[0]); i++) {
        std::printf("%d", array1[i]);

        if (i != sizeof(array1) / sizeof(array1[0]) - 1) {
            std::printf(", ");
        }
    }
    std::printf("}");
}
